pkg/controller/rbac.go

```diff
@@ -65,6 +67,81 @@ func (c *StashController) ensureSidecarRoleBindingDeleted(resource metav1.Object
 		Delete(c.getSidecarRoleBindingName(resource.Name), &metav1.DeleteOptions{})
 }
 
+func (c *StashController) ensureCronJobRBAC(resource *core.ObjectReference) error {
+	if err := c.ensureCronJobRole(resource); err != nil {
+		return err
+
+	}
+	if err := c.ensureCronJobRoleBinding(resource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *StashController) ensureCronJobRole(resource *core.ObjectReference) error {
```

Comment:
1. Use `ClusterRole`.
2. Ensure when operator start.
3. Set Stash operator deployment as owner of this ClusterRole.
	- Get operator deployment name from `MY_POD_NAME` env.
	- Get operator deployment namespace from `MY_POD_NAMESPACE`.
	- Then get deployment using client-go.
	- Finally, set set that deployment as owner of this `ClusterRole`
You should also ensure this `ClusterRole` from where cronjob is created. In case, the cluster can be deleted accidentally.

Reply:
resolved